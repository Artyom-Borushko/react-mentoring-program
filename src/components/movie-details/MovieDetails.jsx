import React from "react";
import './movieDetails.css';

export default function MovieDetails({movie}) {

    return (
        <div className={'movie-details'}>
            <img
                src={movie.src}
                alt='movie details'
                className={'movie-details-image'}
                aria-label={'movie details'}
            />

             <div className={'movie-details-information'}>
                 <div className={'movie-details-title-rating'}>
                     <p className={'movie-details-title'}>{movie.title}</p>
                     <p className={'movie-details-rating'}>{movie.rating}</p>
                 </div>
                 <p className={'movie-details-genres'}>{movie.relevantGenres.join(', ')}</p>

                 <div className={'movie-details-year-duration'}>
                     <span className={'movie-details-year'}>
                         {movie.releaseYear}
                     </span>
                     <span className={'movie-details-duration'}>
                         {movie.duration}
                     </span>
                 </div>

                 <p className={'movie-details-description'}>
                     {movie.description}
                 </p>
             </div>
        </div>
    );
}
