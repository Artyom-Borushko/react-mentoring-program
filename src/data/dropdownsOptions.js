export const movieDropdownOptions = ['edit', 'delete'];

export const moviesSortDropdownOptions = ['release date', 'title'];
