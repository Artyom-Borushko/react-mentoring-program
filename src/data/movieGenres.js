
export const listOfGenres = ['all', 'documentary', 'comedy', 'horror', 'crime'];
