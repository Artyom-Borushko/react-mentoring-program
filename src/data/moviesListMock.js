export const listOfMoviesMock = [
    { src: '/images/movies-list/pulpFiction.png', title: "Pulp Fiction", releaseYear: "2004", relevantGenres: ['action', 'crime'], rating: '9.2', duration: '2h 22min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId1" },
    { src: '/images/movies-list/bohemianRhapsody.png', title: "Bohemian Rhapsody", releaseYear: "2003", relevantGenres: ['documentary', 'drama'], rating: '8.4', duration: '1h 27min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId2" },
    { src: '/images/movies-list/killBillVol2.png', title: "Kill Bill: Vol 2", releaseYear: "1994", relevantGenres: ['comedy', 'crime', 'oscar'], rating: '6.2', duration: '3h 12min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId3" },
    { src: '/images/movies-list/avengersWarOfInfinity.png', title: "Avengers: War of Infinity", releaseYear: "2004", relevantGenres: ['action', 'adventure'], rating: '8.2', duration: '0h 56min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId4" },
    { src: '/images/movies-list/inception.png', title: "Inception", releaseYear: "2003", relevantGenres: ['action', 'adventure', 'horror'], rating: '10.0', duration: '1h 39min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId5" },
    { src: '/images/movies-list/reservoirDogs.png', title: "Reservoir dogs", releaseYear: "1994", relevantGenres: ['crime', 'oscar'], rating: '7.7', duration: '2h 00min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId6" },
];
