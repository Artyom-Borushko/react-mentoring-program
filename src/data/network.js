export default 'http://localhost:4000/movies';
