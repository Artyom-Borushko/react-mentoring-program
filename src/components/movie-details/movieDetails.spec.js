import {render} from "@testing-library/react";
import React from "react";
import '@testing-library/jest-dom';
import {listOfMoviesMock} from "../../data/moviesListMock";
import MovieDetails from "./MovieDetails";

describe("MovieDetails component tests", () => {

    it("renders MovieDetails component", () => {
        const {container} = render(<MovieDetails
            movie={listOfMoviesMock[0]}
        />);

        expect(container).toMatchSnapshot();
    });
});
