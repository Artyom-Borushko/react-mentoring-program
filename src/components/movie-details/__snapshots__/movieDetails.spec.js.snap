// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component tests renders MovieDetails component 1`] = `
<div>
  <div
    class="movie-details"
  >
    <img
      alt="movie details"
      aria-label="movie details"
      class="movie-details-image"
      src="/images/movies-list/pulpFiction.png"
    />
    <div
      class="movie-details-information"
    >
      <div
        class="movie-details-title-rating"
      >
        <p
          class="movie-details-title"
        >
          Pulp Fiction
        </p>
        <p
          class="movie-details-rating"
        >
          9.2
        </p>
      </div>
      <p
        class="movie-details-genres"
      >
        action, crime
      </p>
      <div
        class="movie-details-year-duration"
      >
        <span
          class="movie-details-year"
        >
          2004
        </span>
        <span
          class="movie-details-duration"
        >
          2h 22min
        </span>
      </div>
      <p
        class="movie-details-description"
      >
        Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!
      </p>
    </div>
  </div>
</div>
`;
